solutions/devsprint-eduardo-bocato-1/DesignSystem/DS+MaxLines.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  DS+Size.swift
+//  SpaceApp
+//
+//  Created by Razee Hussein-Jamal on 10/03/2022.
+//
+
+import CoreGraphics
+
+extension DS {
+    public struct MaxLines {
+        let value: Int
+    }
+}
+
+extension DS.MaxLines {
```

Comment:
**[SUGGESTION]**
In the case of max lines, this is directly related to a Typography or style, the intention here is great, but there is room for improvement...

```swift
extension DS {
    public struct Typography {
        let font: Font
        let maxLines: Int
        
        init(font: Font, maxLines: Int = .zero) {
            self.font = font
            self.maxLines = maxLines
        }
    }
}

extension DS.Typography {
    public static let descriptionLabel: Self = .init(font: .body, maxLines: 3)
}

public extension Text {
    func dsTypography(_ typography: DS.Typography) -> some View {
        self.font(typography.font)
    }
    
    func dsTypography(
        _ typography: DS.Typography,
        color: DS.Color
    ) -> some View {
        self
            .font(typography.font)
            .lineLimit(typography.maxLines)
            .dsForegroundColor(color)
    }
}

// Then, on the view...
Text("something")
    .dsTypography(.descriptionLabel)
```

Reply:
Thank you, this looks much better